src/document.jl

```diff
@@ -90,6 +90,13 @@ function parse_file(filename::AbstractString)
     XMLDocument(p)
 end
 
+function parse_file(filename::AbstractString, encoding, options::Integer)
+    p = ccall((:xmlReadFile,libxml2), Xptr, (Cstring, Ptr{Cchar}, Cint),
```

Comment:
Oh, I see, you want to be able to pass `C_NULL` for `encoding`.  That seems pretty un-Julian to expose.  Maybe just have a separate `parse_file(filename, options)` method?